src/uct/ib/dc/dc_mlx5_ep.c

```diff
@@ -1245,13 +1245,20 @@ uct_dc_mlx5_iface_dci_do_pending_wait(ucs_arbiter_t *arbiter,
     ucs_assert(!uct_dc_mlx5_iface_is_dci_rand(iface));
 
     if (ep->dci != UCT_DC_MLX5_EP_NO_DCI) {
+        /* in case if group is not empty and no outstanding OP's then
+         * force schedule EP to TX waiting queue */
+        if (!uct_dc_mlx5_iface_dci_has_outstanding(iface, ep->dci) &&
```

Comment:
why need to check uct_dc_mlx5_iface_dci_has_outstanding() ?

Reply:
in case if there are outstanding ops then EP will be scheduled in poll_tx function via dci_put

Reply:
1. it's weird that we have ep on this arbiter with valid DCI, how was it allocated?
2. even if we have tx outstanding, we should schedule the dci to tx-wait arbiter if it got a valid dci somehow
3. is it really possible pending will be empty? since this function is processing a pending element

Reply:
when CHECK_RES macro is called it is trying to allocate DCI
not sure about empty pending queue... removed evaluation